WordPress/gutenberg
# Separator block: refactor to use color block supports

## Description
The separator block is currently using its own color control to set a custom color attribute, which means it can't take advantage of the alpha channel support in the  color block supports color picker in order to [fix this issue](https://github.com/WordPress/gutenberg/issues/34637).

This PR refactors the block to use the new color block support.

## Testing Instructions

- In a block based theme add a Separator block and make sure all the options work in editor and frontend, including setting the alpha channel in the color picker
- Make sure blocks created with all options migrate successfully when opened in editor running on this branch.
- Check that the v1 blocks with color get a default 0.4 opacity applied, when unmigrated and viewed in the frontend, and also when loaded in the editor and resaved. If the blocks color is edited the default opacity should be removed.

<details><summary>V1 Block code examples</summary>

```html

<hr class="wp-block-separator"/>



<hr class="wp-block-separator is-style-wide"/>



<hr class="wp-block-separator is-style-dots"/>



<hr class="wp-block-separator has-text-color has-background has-vivid-red-background-color has-vivid-red-color"/>



<hr class="wp-block-separator has-text-color has-background" style="background-color:#ff0000;color:#ff0000"/>



<hr class="wp-block-separator has-text-color has-background has-vivid-red-background-color has-vivid-red-color is-style-dots"/>



<hr class="wp-block-separator has-text-color has-background is-style-dots" style="background-color:#ff0000;color:#ff0000"/>

```
</details>

## Screenshots 
![separator](https://user-images.githubusercontent.com/3629020/152472024-716c2c84-d43a-43d3-89ee-28ad850c97c7.gif)

## Backwards compatibility considerations

As well as ensuring that the 0.4 opacity was maintained for both static content, and content that has been opened in the editor, the HTML structure and classnames were checked between old and new block versions. 

The HTML structure of saved content remains the same, with just the addition of a new classname to cover the opacity backwards compatibility. This change should have no impact on themes and plugins that are applying custom CSS via the existing classes and elements.

<img width="1294" alt="Screen Shot 2022-02-23 at 4 07 17 PM" src="https://user-images.githubusercontent.com/3629020/155255545-990e0c0e-6878-4757-9bf0-28668cd32bf7.png">
